### PR TITLE
Core: Add flag to control sending metric reports via REST

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java [hunk 1]
```diff
@@ -316,14 +324,18 @@ private void reportMetrics(
...
+      reporter.report(report);
+      if (reportingViaRestEnabled) {
+        client.post(
+            paths.metrics(tableIdentifier),
+            ReportMetricsRequest.of(report),
+            null,
+            headers,
+            ErrorHandlers.defaultErrorHandler());
+      } else {
+        LOG.debug("Reporting metrics to REST endpoint is disabled");
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java:337]: I don't think this is a helpful debug message every time a report may go out. There isn't anything specific to the report here. I'd rather remove this. If we want to have a debug log for config properties, then we can log when we set `reportingViaRestEnabled`.

**Comment by rdblue** [comment]
Looks like this also adds support for the metrics-impl property. Does this replace the other PR?

File: core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java (hunk 1 above)
Comment (nastra) [reply core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java]: fair enough, I've removed the debug logging here and rebased the PR

**Comment by nastra** [comment]
@rdblue I've rebased the PR, so the old commits with the metrics-impl property are gone (since they are merged already).

**Review by rdblue — APPROVED** [review]

**Pull request merged**